### PR TITLE
Feature/cve severity

### PR DESCRIPTION
**Pull request by ghost**
Checks various CVE security levels for a new release, and alerts differently based on them

### CODE REVIEW

File: health.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+	}
+}
+
+func checkCriticalSecurityScore(repo *releaseRepository) func() (string, error) {
+	return func() (string, error) {
+		repo.RLock()
+		defer repo.RUnlock()
+
+		_, err := compareInstalledWithLatest(repo)()
+		if err != nil && *repo.latestVersion.MaxCVSS > 9 {
```
Comment (kghatton) [inline health.go]: Do you mean >=9

File: health.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+func checkHighSecurityScore(repo *releaseRepository) func() (string, error) {
+	return func() (string, error) {
+		repo.RLock()
+		defer repo.RUnlock()
+
+		_, err := compareInstalledWithLatest(repo)()
+		if err != nil &&
+			repo.latestVersion.MaxCVSS != nil &&
+			*repo.latestVersion.MaxCVSS > 7 &&
```
Comment (kghatton) [inline health.go]: Again, I think this should be >=

File: release_test.go [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+)
+
+func TestCoreOS(t *testing.T) {
+	repo := newReleaseRepository(&http.Client{})
+	coreOS, err := repo.Get("1284.2.0")
+	assert.NoError(t, err)
+
+	d, _ := json.Marshal(coreOS)
+	t.Log(string(d))
+}
```
Comment (kghatton) [inline release_test.go]: I think you're missing some test cases here.

**Review by kghatton — CHANGES_REQUESTED** [review]

File: circle.yml [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+test:
+  pre:
+    - go get -u github.com/jstemmer/go-junit-report
+    - go get -u github.com/mattn/goveralls
+    - cd $PROJECT_PATH && wget https://raw.githubusercontent.com/Financial-Times/cookiecutter-upp-golang/master/coverage.sh && chmod +x coverage.sh
+  override:
+    - mkdir -p $CIRCLE_TEST_REPORTS/golang
+    - cd $PROJECT_PATH && govendor test -race -v +local | go-junit-report > $CIRCLE_TEST_REPORTS/golang/junit.xml
+    - cd $PROJECT_PATH && ./coverage.sh
```
Comment (kghatton) [inline circle.yml:30]: Could you add a `docker build` command here, so that the Dockerfile is tested?

**Review by kghatton — COMMENTED** [review]
I've suggested an addition to the Circle setup, otherwise this looks fine.

File: circle.yml (hunk 4 above)
Comment (ghost) [reply circle.yml:30]: Added. :)

**Review by kghatton — APPROVED** [review]

File: util.go [hunk 5]
```diff
@@ -1,29 +1,49 @@
...
+	req, err := http.NewRequest("GET", uri, nil)
+	if err != nil {
+		return nil, err
 	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
```
Comment (tosan88) [inline util.go:39]: Could you please also defer closing the response body?

File: Dockerfile [hunk 6]
```diff
@@ -1,16 +1,25 @@
...
+  && go build  \
+  && mv ${PROJECT} /${PROJECT} \
+  && apk del .build-dependencies \
   && rm -rf $GOPATH /var/cache/apk/*
 
+WORKDIR /
+
 EXPOSE 8080
-CMD ["/coreos-version-checker"]
+CMD [ "/concordance-rw-dynamodb" ]
```
Comment (tosan88) [inline Dockerfile:25]: Not the correct one

File: main.go [hunk 7]
```diff
@@ -1,36 +1,92 @@
...
+	coreOSReleaseConfPath *string
+)
+
+const (
+	versionUri string = "http://%s.release.core-os.net/amd64-usr/current/version.txt"
 )
 
 func main() {
-	app := cli.App("CoreOS-version-checker", "A service that report on current VM status at __health")
+	app := cli.App("coreos-version-checker", "Checks for new CoreOS upgrades, and reports on the CVE severity score.")
```
Comment (tosan88) [inline main.go:23]: Also maybe it's worth setting the log formatter to be JSON

**Pull request merged**